libtelemetry/src/full/java/com/mapbox/android/telemetry/provider/MapboxTelemetryInitProvider.java

```diff
@@ -0,0 +1,78 @@
+package com.mapbox.android.telemetry.provider;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.pm.ProviderInfo;
+import android.database.Cursor;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.mapbox.android.telemetry.location.LocationCollectionClient;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.mapbox.android.telemetry.location.LocationCollectionClient.DEFAULT_SESSION_ROTATION_INTERVAL_HOURS;
+
+public class MapboxTelemetryInitProvider extends ContentProvider {
```

Comment:
Why do we need a ContentProvider here? Just for init?

Reply:
yes, it's a hacky way to execute your code at app startup